src/libstd/num/f32.rs

```diff
@@ -120,13 +190,73 @@ impl Float for f32 {
     #[allow(deprecated)]
     #[inline]
     fn max_value() -> f32 { num::Float::max_value() }
-
+    /// Returns `true` if the number is `NaN`.
+    ///
+    /// ```
+    /// use std::f32;
+    ///
+    /// let nan: f32 = f32::NAN;
+    ///
+    /// assert!(nan.is_nan());
+    /// assert!(!7.0f32.is_nan());
+    /// ```
     #[inline]
     fn is_nan(self) -> bool { num::Float::is_nan(self) }
+    /// Returns `true` if the number is infinite.
+    ///
+    /// ```
+    /// use std::Float;
+    /// use std::f32;
+    ///
+    /// let inf: f32 = Float::infinity();
+    /// let neg_inf: f32 = Float::neg_infinity();
+    /// let nan: f32 = f32::NAN;
+    ///
+    /// assert!(!7.0f32.is_infinite());
+    /// assert!(!nan.is_infinite());
+    ///
+    /// assert!(inf.is_infinite());
+    /// assert!(neg_inf.is_infinite());
+    /// ```
     #[inline]
     fn is_infinite(self) -> bool { num::Float::is_infinite(self) }
+    /// Returns `true` is the number is finite.
+    ///
+    /// ```
+    /// use std::Float;
+    /// use std::f32;
+    ///
+    /// let inf: f32 = Float::infinity();
+    /// let neg_inf: f32 = Float::neg_infinity();
+    /// let nan: f32 = f32::NAN;
+    ///
+    /// assert!(7.0f32.is_finite());
+    ///
+    /// assert!(!nan.is_finite());
+    /// assert!(!inf.is_finite());
+    /// assert!(!neg_inf.is_finite());
+    /// ```
     #[inline]
     fn is_finite(self) -> bool { num::Float::is_finite(self) }
+    /// Returns `true` if the number is neither zero, infinite, subnormal, or NaN.
+    ///
+    /// ```
+    /// use std::num::Float;
+    /// use std::f32;
+    ///
+    /// let min: f32 = f32::MIN_POS_VALUE; // 1.17549435e-38f32
+    /// let max: f32 = f32::MAX_VALUE;
+    /// let lower_than_min: f32 = 1.0e-40;
```

Comment:
Should provide a link...less than min is weird. This could be puzzling if you've never seen this before. I hadn't.